akka-actor-typed-tests/src/test/scala/akka/actor/typed/ActorRefIgnoreSpec.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2019-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import akka.actor.typed.scaladsl.Behaviors
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import akka.actor.testkit.typed.scaladsl.ActorTestKit
+import org.scalatest.BeforeAndAfterAll
+
+class ActorRefIgnoreSpec extends AnyWordSpec with ScalaFutures with Matchers with BeforeAndAfterAll {
+
+  case class Pong(counter: Int)
+  case class Ping(replyTo: ActorRef[Pong])
+
+  def pingPongBehavior(counter: Int): Behavior[Ping] =
+    Behaviors.receiveMessage[Ping] {
+      case Ping(replyTo) =>
+        val newCounter = counter + 1
+        replyTo ! Pong(newCounter)
+        pingPongBehavior(newCounter)
+    }
+
+  val testKit = ActorTestKit()
+
+  override def afterAll(): Unit = testKit.shutdownTestKit()
+
+  "ActorRef.ignore" should {
+
+    "return an ActorRef that ignore all incoming messages" in {
+      val pingPongRef = testKit.spawn(pingPongBehavior(0))
+
+      val probe = testKit.createTestProbe[Pong]("pong")
+      pingPongRef ! Ping(probe.ref)
+      probe.expectMessage(Pong(1))
+
+      // this is more a compile-time proof
+      // since the reply is ignored, we can't check that a message was sent to it
+      pingPongRef ! Ping(ActorRef.ignore)
+
+      // but we do check that the counter has increased when we sent the ActorRef.ignore
+      pingPongRef ! Ping(probe.ref)
+      probe.expectMessage(Pong(3))
+    }
+  }
```

Comment:
Some unittytesty things perhaps - will two `ignore` compare the same?

I think we should also cover some more corner cases here, if nothing else just to document expected behavior:

What if the recipient (unknowingly that it is ignore):
 * tries to ask the ref something (response-with-ack)
 * tries to watch the ref
 * adapt to classic and use that as sender in interaction with classic actor - which leads to
   * future.pipeTo(ignore) 
   * classic forward(ignore)
   * classic router.route
 * any more I'm not coming up with from the top of my head?

Reply:
Tests for ask and watch was added. Also conversion to classic back and forth. 

As discussed, we are postponing the decision to add more tests to cover integration with Akka classic.

Reply:
Any other tests ideas or we can consider this one as resolved for now?

Reply:
Looks good enough for me